test/files/run/t9644.scala

```diff
@@ -9,7 +9,7 @@ object Test extends App {
 
   import reflect.runtime.universe._
 
-  val List(d, r) = symbolOf[Foo].annotations
+  val List(r, d) = symbolOf[Foo].annotations
```

Comment:
Is `r, d` the order in the pickle? Where is the order being reversed (compared to the source file)?

Reply:
I added a test for separate compilation, demonstrating the same order. Because I mostly use a landscape-mode monitor with a text editor, I didn't notice that in the presence of a scala sig, we don't see other runtime attributes. Also, I see that on the emitting end, we just set a flag, so I assume asm emits the attribute, which is present in the Scala class file in the test. Probably, adding JavaDeprecatedAttr in class file parser was deemed duplicative; mental Edit: not because of Scala deprecated but because of the runtime annotation. The whole point of this exercise is that we can't rely on the runtime annotation (which scalac itself does not emit, but possibly should, so that Java tooling would see since value written in Scala). Further mental note: Reviewing the long history (2014-2020) of trying to handle java annotations, it's hard to see what worked when. I agree with the original lrytz ticket advocating uniformity.

I probably misunderstood the hrhino comment about order, since the commit says compile order is different. Does order mean source order or insertion order? JVMS does not suggest that attributes or annotations have a natural or unnatural order.

Order could matter, such as `@nowarn` interleaved with buggy annotations. (Hypothetically.)